agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows_test.go

```diff
@@ -0,0 +1,700 @@
+	ssmClientCreator := mock_ssm_factory.NewMockSSMClientCreator(ctrl)
+	asmClientCreator := mock_asm_factory.NewMockClientCreator(ctrl)
+	fsxClientCreator := mock_fsx_factory.NewMockFSxClientCreator(ctrl)
+	fsxWindowsFileServerRes := &FSxWindowsFileServerResource{
+		knownStatusUnsafe:   resourcestatus.ResourceCreated,
+		desiredStatusUnsafe: resourcestatus.ResourceCreated,
+	}
+	fsxWindowsFileServerRes.Initialize(&taskresource.ResourceFields{
+		ResourceFieldsCommon: &taskresource.ResourceFieldsCommon{
+			SSMClientCreator:   ssmClientCreator,
+			ASMClientCreator:	asmClientCreator,
+			FSxClientCreator:	fsxClientCreator,
+			CredentialsManager: credentialsManager,
+		},
+	}, apitaskstatus.TaskStatusNone, apitaskstatus.TaskRunning)
+
+	assert.NotNil(t, fsxWindowsFileServerRes.credentialsManager)
+	assert.NotNil(t, fsxWindowsFileServerRes.ssmClientCreator)
+	assert.NotNil(t, fsxWindowsFileServerRes.asmClientCreator)
+	assert.NotNil(t, fsxWindowsFileServerRes.fsxClientCreator)
+	assert.NotNil(t, fsxWindowsFileServerRes.statusToTransitions)
+}
+
+func TestMarshalUnmarshalJSON(t *testing.T) {
+	volumeConfig := FSxWindowsFileServerVolumeConfig{
+		FileSystemID:                 "fs-12345678",
+		RootDirectory:                rootDirectory,
+		AuthConfig:                   FSxWindowsFileServerAuthConfig{
+			CredentialsParameter: credentialsParameter,
+			Domain: domain,
+		},
+		FSxWindowsFileServerHostPath: hostPath,
+	}
+	fsxWindowsFileServerIn := &FSxWindowsFileServerResource{
+		Name:                        "test",
+		VolumeConfig:                volumeConfig,
+		taskARN:                     taskARN,
+		executionCredentialsID:      executionCredentialsID,
+		createdAtUnsafe:             time.Time{},
+		knownStatusUnsafe:           resourcestatus.ResourceCreated,
+		desiredStatusUnsafe:         resourcestatus.ResourceCreated,
+	}
+
+	bytes, err := json.Marshal(fsxWindowsFileServerIn)
+	require.NoError(t, err)
+
+	fsxWindowsFileServerOut := &FSxWindowsFileServerResource{}
+	err = json.Unmarshal(bytes, fsxWindowsFileServerOut)
+	require.NoError(t, err)
+	assert.Equal(t, fsxWindowsFileServerIn.Name, fsxWindowsFileServerOut.Name)
+	assert.Equal(t, fsxWindowsFileServerIn.VolumeConfig, fsxWindowsFileServerOut.VolumeConfig)
+	assert.Equal(t, fsxWindowsFileServerIn.taskARN, fsxWindowsFileServerOut.taskARN)
+	assert.Equal(t, fsxWindowsFileServerIn.executionCredentialsID, fsxWindowsFileServerOut.executionCredentialsID)
+	assert.WithinDuration(t, fsxWindowsFileServerIn.createdAtUnsafe, fsxWindowsFileServerOut.createdAtUnsafe, time.Microsecond)
+	assert.Equal(t, fsxWindowsFileServerIn.desiredStatusUnsafe, fsxWindowsFileServerOut.desiredStatusUnsafe)
+	assert.Equal(t, fsxWindowsFileServerIn.knownStatusUnsafe, fsxWindowsFileServerOut.knownStatusUnsafe)
+}
+
+func TestRetrieveCredentials(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	credentialsManager := mock_credentials.NewMockManager(ctrl)
```

Comment:
Instead of using similar setup structures
Can you move the test setup to a helper method and re-use it?

Reply:
will do